### PR TITLE
Marathon endpoint for `pod kill` now ends with plural `::instances`

### PR DESCRIPTION
**Pull request by cruhland**
Pod tests for PR #806 won't pass until this is merged.

### CODE REVIEW

**Comment by cruhland** [comment]
@tamarrow I made some significant test changes, please take a look!

File: cli/tests/integrations/test_task.py [hunk 1]
```diff
@@ -328,3 +328,15 @@ def _get_completed_task_id(app_id='test-app-completed'):
     assert len(tasks) == 1
     task_id = tasks[0]['id']
     return task_id
+
+
+def _read_lines_once_available(file_obj):
```
Comment (tamarrow) [inline cli/tests/integrations/test_task.py]: rename to `_read_lines(file_obj)`, we don't need to expose implementation. Please add a pydoc :)

File: cli/tests/integrations/common.py [hunk 2]
```diff
@@ -701,3 +701,41 @@ def base64_to_dict(byte_string):
...
+    assert_command(
+        ['dcos', 'package', 'repo', 'remove', 'Universe'])
+
+    assert_command(
+        ['dcos', 'package', 'repo', 'add', 'test-universe', UNIVERSE_TEST_REPO]
+    )
+
+    # Give the test universe some time to become available
+    describe_command = ['dcos', 'package', 'describe', 'helloworld']
+    for _ in range(10):
```
Comment (tamarrow) [inline cli/tests/integrations/common.py]: the problem is sometimes this takes > 10 seconds.

File: cli/tests/integrations/common.py [hunk 3]
```diff
@@ -701,3 +701,41 @@ def base64_to_dict(byte_string):
...
+        returncode, _, _ = exec_command(describe_command)
+        if returncode == 0:
+            break
+        time.sleep(1)
+    else:
+        teardown_universe_server()
```
Comment (tamarrow) [inline cli/tests/integrations/common.py]: this call is unnecessary. should be done by the teardown module

Comment (cruhland) [inline cli/tests/integrations/common.py]: Nope! When the assertion is raised on the following line, PyTest does not call `teardown_module()` afterwards. This happened to me with local testing and I had to manually run the teardown commands. The library must assume that the setup code will not raise an exception.

File: cli/tests/integrations/common.py (hunk 2 above)
Comment (cruhland) [inline cli/tests/integrations/common.py]: I'll bump it up to 30 seconds. If it's slower than that, something must be really wrong. And if it's faster than that, this code won't wait for the full time anyway.

File: cli/tests/integrations/common.py (hunk 3 above)
Comment (cruhland) [inline cli/tests/integrations/common.py]: I'll add a comment to clarify.

Comment (tamarrow) [inline cli/tests/integrations/common.py]: please link to documentation

**Pull request merged**